packages/plugins/typescript-resolvers/tests/ts-resolvers.spec.ts

```diff
@@ -680,6 +680,85 @@ describe('TypeScript Resolvers Plugin', () => {
     validateTs(content);
   });
 
+  it('should use {} as default of rootValueType', async () => {
+    const testSchema = buildSchema(/* GraphQL */ `
+      type Subscription {
+        postAdded: Post
+      }
+
+      type Query {
+        posts: [Post]
+      }
+
+      type Mutation {
+        addPost(author: String, comment: String): Post
+      }
+
+      type Post {
+        author: String
+        comment: String
+      }
+    `);
+    const content = await plugin(testSchema, [], {}, { outputFile: 'graphql.ts' });
+
+    expect(content).toBeSimilarStringTo(`
+      export type ResolversTypes = {
+        Query: {},
+        Post: Post,
+        String: Scalars['String'],
+        Mutation: {},
+        Subscription: {},
+        Boolean: Scalars['Boolean'],
+      };
+    `);
+  });
+
+  it('should use rootValueType in Query, Mutation and Subscription', async () => {
+    const testSchema = buildSchema(/* GraphQL */ `
+      type MySubscription {
+        postAdded: Post
+      }
+
+      type MyQuery {
+        posts: [Post]
+      }
+
+      type MyMutation {
+        addPost(author: String, comment: String): Post
+      }
+
+      type Post {
+        author: String
+        comment: String
+      }
+
+      schema {
+        query: MyQuery
+        mutation: MyMutation
+        subscription: MySubscription
+      }
+    `);
+    const content = await plugin(
+      testSchema,
+      [],
+      {
+        rootValueType: 'MyRoot',
```

Comment:
That's the idea of `rootValueType`, it's the same for all root types and it's something different than `context` or even a `mapper` (it shouldn't be wrapped with `Partial` and it's the same signature for Query, Mutation and Subscription - using mapper will require to define the type for all three objects).